### PR TITLE
Add java templates

### LINKED ISSUE #1016: Add new java templates to `azd template list`
Similar change - https://github.com/Azure/azure-dev/pull/610

### PR DESCRIPTION
**Pull request by weikanglim**
Add java templates to `azd` templates list.

Display templates sorted by name. This groups languages together and provides better consistency:
image

image

Fixes #1016

### CODE REVIEW

**Review by vhvb1989 — COMMENTED** [review]
Do we have a test to check they stay sorted in the future (to prevent something change this )

We might be able to test the action directly and test the ordering

**Review by vhvb1989 — APPROVED** [review]

**Comment by weikanglim** [comment]
> Do we have a test to check they stay sorted in the future (to prevent something change this )
> 
> We might be able to test the action directly and test the ordering

I was thinking along the lines of the same thing.. Mostly was trying to avoid it because there are still some remaining issues for testing actions. It was insightful to try to write the test. I think we definitely should move into the direction of having `action.Run` return a result object, and create command-layer tests for contracts.

I think that can also solve your problem with `PostRun` messages.

**Comment by weikanglim** [comment]
I might need to hold this off from being checked in until the repositories are made public

**Comment by weikanglim** [comment]
> /azp run azure-dev - template - tests

@rajeshkamal5050 FYI this still runs against main currently. I'm fixing `template-tests` to finally allow us to run this in-PR

File: cli/azd/cmd/templates.go [hunk 1]
```diff
@@ -81,10 +83,10 @@ func (tl *templatesListAction) Run(ctx context.Context) error {
...
 	}
 
-	templateList := make([]templates.Template, 0, len(templateSet))
-	for _, template := range templateSet {
-		templateList = append(templateList, template)
-	}
+	templateList := maps.Values(templateSet)
+	slices.SortFunc(templateList, func(a, b templates.Template) bool {
+		return a.Name < b.Name
+	})
```
Comment (wbreza) [inline cli/azd/cmd/templates.go:89]: Any specific reason to use `slices.SortFunc(...)` over `sort.Slice(...)` ?

File: eng/pipelines/template-tests.yml [hunk 2]
```diff
@@ -45,38 +117,16 @@ jobs:
       name: azsdk-pool-mms-ubuntu-2004-general
       vmImage: MMSUbuntu20.04
 
+    container: $[ variables['AzdContainerImage'] ]
+
     steps:
       - checkout: self
 
-      - template: /eng/pipelines/templates/steps/install-bicep.yml
```
Comment (wbreza) [inline eng/pipelines/template-tests.yml:51]: Why are we removing these CLI installations?

**Review by wbreza — APPROVED** [review]
A couple minor questions, but otherwise looks good.

File: eng/pipelines/template-tests.yml (hunk 2 above)
Comment (weikanglim) [reply eng/pipelines/template-tests.yml]: I was trying to move us towards using our published dev containers. I think that completes the end-to-end verification loop for adding a template in `azd`. Change has been reverted and will come in as a separate PR.

File: cli/azd/cmd/templates.go (hunk 1 above)
Comment (weikanglim) [reply cli/azd/cmd/templates.go:89]: `sort.Slice` looks like the old-non-generic implementation. `slices.SortFunc` is the newer generic-based implementation. They both work the same for our use case.

**Comment by weikanglim** [comment]
Successful template-tests run: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1956479&view=results

**Pull request merged**